src/router/AppRouter.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect, useState } from "react";
+import { Routes, Route, useNavigate, Navigate } from "react-router-dom";
+import { UserForm } from "../pages/Form";
+import { Dashboard } from "../pages/Dashboard";
+import { Layout } from "../shared/Layout";
+import { useAuth } from "../hooks/useAuth";
+import { AddVacation } from "../pages/AddVacation";
+
+const AppRouter = () => {
+  const { user, loading } = useAuth();
+  const [isBackButton, setIsBackButton] = useState<boolean>(false);
+  const [title, setTitle] = useState<string>("Dashboard");
+
+  if (loading) return <div>loading</div>;
```

Comment:
Loading here is not needed. It's handled on the components level.
Why do you think it should be here?